torch/csrc/jit/codegen/cuda/scheduler/pointwise.h

```diff
@@ -10,6 +10,130 @@ namespace jit {
+ * running at the same time on the device we can consider a wave on the GPU of
+ * incrementing BIDx (wave number of times), while holding TIDy constant BDIMy *
+ * wave number of times.
+ *
+ * If instead we have a situation like:
+ * T0[i0, i1, b2]
+ * T1[i0, b1, i2]
+ * T2[i0, i1, i2] = T0 + T1
+ * It makes sense that the break point would be in position 2, between i1 and
+ * i2. This is because when we map [i0, i1 | i2] to [BIDy, TIDy| BIDx, TIDx]
+ * BIDx, and TIDx will access the same elements of T1, and TIDy will likely
+ * access the same elements of T0 (as long as i1 > BDIMy). Even if i1 on the
+ * order of BDIMy we'll only use two unique elemen*ts per increment of BIDx or
+ * TIDx. This means we'll still reuse many of the same values and limit the
+ * amount we need to reread values in T0 and T1.
+ *
+ * If instead we have:
+ * T0[i0, b1, i2]
+ * T1[b0, i1, i2]
+ * T2[i0, i1, i2] = T0 + T1
+ * The analysis gets a bit more complicated. First if i2 is very large and i0
+ * and i1 are relatively small it would make sense to have [i0, i1 | i2]. If b0
+ * is very small it's unlikely beneficial to have [i0 | i1, i2]. If i2 is very
+ * small it may be worthwhile to have [i0 | i1, i2]. If i1 and i2 are not small,
+ * and their product is relatively large (i.e. you can't fit T2[i, :, :] in L2)
+ * then it's unlikely we'll get any significant reuse across i0.
+ *
+ * What we should assume then, is that we will always get strong reuse across
+ * TIDy by a factor of BDIMy, but for reuse across BIDx and TIDx we'll only get
+ * good reuse for broadcasts that are immediately and consecutively left of the
+ * break point.
+ *
+ * For example if we have:
+ * T0[i0, b1, i2]
+ * T1[b0, b1, i2]
+ * T2[b0, i1, i2]
+ * T3[i0, i1, i2] = T0 + T1 + T2
+ * We may want to break point at position 1 or position 2 (i.e. [i0 | i1, i2] or
+ * [i0, i1 | i2]). We can't immediately tell from the structure.
+ *
+ * If we choose [i0, i1 | i2] then we'll get strong reuse of T0 on TIDy, we'll
+ * get perfect reuse across T1 on TIDy, likely no reuse on T2.
+ *
+ * If we pick [i0 | i1, i2], we may get some reuse on T0 across BIDx, we'll get
+ * perfect reuse across TIDy and likely some reuse across BIDx on T1, and
+ * perfect reuse across TIDy on T2.
+ *
+ * Materializing these benefits is dependent on the decisions the scheduler
+ * makes when parallelizing the problem. The heuristics logic at the moment is
+ * fairly simplistic where it assumes that there's only reuse across the break
+ * points for tensors that have no iteration domain on the entire side of the
+ * breakpoint. This is not optimal but for the time being it seems sufficient.
+ * We would ideally take into consideration the parallelization scheme and
+ * partial broadcasting on the lhs or rhs.
+ *
+ * An example of how this analysis is done is given the DAG:
+ * T0[i0, i1, b2] float
+ * T1[i0, b1, i2] half
+ * T2[i0, b1, i2] = cast(T1, float)
+ * T4[i0, i1, i2] float = T0 + T1
```

Comment:
I think you meant the following?
```suggestion
 * T4[i0, i1, i2] float = T0 + T2
```